kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/ExceptionHandlingTableWriter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.channel;
+
+import io.tabular.iceberg.connect.WriteExceptionHandler;
+import io.tabular.iceberg.connect.data.Utilities;
+import io.tabular.iceberg.connect.data.WriterResult;
+import java.util.List;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+class ExceptionHandlingTableWriter implements TableWriter, AutoCloseable {
+
+  private final TableWriter underlying;
+  private final WriteExceptionHandler exceptionHandler;
+
+  ExceptionHandlingTableWriter(TableWriter underlying, WriteExceptionHandler exceptionHandler) {
+    this.underlying = underlying;
+    this.exceptionHandler = exceptionHandler;
+  }
+
+  @Override
+  public void write(SinkRecord sinkRecord, String tableName, boolean ignoreMissingTable) {
+    try {
+      underlying.write(sinkRecord, tableName, ignoreMissingTable);
```

Comment:
This is a very broad net here which makes me uncomfortable. 

In my PR there is a reason why I introduced `CatalogAPI` wrapping a bunch of the calls to Iceberg.  Is it schema failing? is it creating the table failing? is it evolving the table failure?  Is it the catalog failing? is it the partitionspec failing?  Much more fine grained control on the errors.  This is catching a ton of errors from many sources and the opportunities for getting it wrong are large.

Reply:
I raise a DeadLetterException wrapping the underlying exception.  Again, it's not that easy/clean when you start putting it all together.